src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -85,7 +86,8 @@ case class AppDefinition(
   def portIndicesAreValid(): Boolean = {
     val validPortIndices = 0 until hostPorts.size
     healthChecks.forall { hc =>
-      validPortIndices contains hc.portIndex
+      (hc.protocol == Protocol.COMMAND && hc.portIndex == 0) ||
```

Comment:
I think it's correct. The ratio: if no portIndex is defined, the default portIndex==0 is set, which is fine (== valid). If portIndex!=0, then a portIndex is explicitly set in the app definition. If this is the case, the portIndex should be valid, i.e. in the index set of the app.